tuf/api/metadata.py

```diff
@@ -307,7 +307,8 @@ def sign(
         """Creates signature over ``signed`` and assigns it to ``signatures``.
 
         Args:
-            signer: A securesystemslib.signer.Signer implementation.
+            signer: Custom or securesystemslib implementation of the
+                ``securesystemslib.signer.Signer`` interface.
```

Comment:
This change does not seem like a huge improvement wrt https://github.com/theupdateframework/python-tuf/pull/1815#pullrequestreview-875726152.

What about
```suggestion
            signer: A ``securesystemslib.signer.Signer`` object that provides a private
                key and signing implementation to generate the signature. A standard
                implementation is available in ``securesystemslib.signer.SSlibSigner``.
```